src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,27 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  def takeRight(n: Int): Iterator[A] = {
+    val buffer = new Array[Any](n)
+    var i = 0
+    var m = 0
+    while (n > 0 && self.hasNext) {
+      buffer.update(i, self.next())
+      i = (i + 1) % n
+      if (m < n) m += 1
+    }
+    val l = m
+    new Iterator[A]() {
+      override def hasNext: Boolean = m > 0
+      override def next(): A = {
+        val value = buffer(i % l).asInstanceOf[A]
+        i = (i + 1) % l
+        m -= 1
+        value
+      }
+    }
+  }
```

Comment:
Oh yes, good point! Then I think your approach sounds good. We should probably override `drop`, `take`, `takeRight` and `dropRight` in `IndexedSeq` for performance. What do others think?